route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -326,25 +281,6 @@ namespace route_following_plugin
         return true;
     }
 
-    void RouteFollowingPlugin::updateCurrentStatus(cav_msgs::Maneuver maneuver, double& speed, double& current_progress, int& lane_id){
-        if(maneuver.type == cav_msgs::Maneuver::STOP_AND_WAIT){
-            speed =0.0;
-            current_progress = maneuver.stop_and_wait_maneuver.end_dist;
-            lane_id = stoi(maneuver.stop_and_wait_maneuver.ending_lane_id);
-        }
-        else if(maneuver.type == cav_msgs::Maneuver::LANE_FOLLOWING){
-            speed =  maneuver.lane_following_maneuver.end_speed;
-            current_progress =  maneuver.lane_following_maneuver.end_dist;
-            lane_id =  stoi(maneuver.lane_following_maneuver.lane_id);
-        }
-        else{
-            speed = GET_MANEUVER_PROPERTY(maneuver,end_speed);
-            current_progress =GET_MANEUVER_PROPERTY(maneuver,end_dist);
-            lane_id = stoi(GET_MANEUVER_PROPERTY(maneuver,ending_lane_id));
-        }
-
-    }
-
     double RouteFollowingPlugin::findSpeedLimit(const lanelet::ConstLanelet& llt)
     {
         lanelet::Optional<carma_wm::TrafficRulesConstPtr> traffic_rules = wm_->getTrafficRules();
```

Comment:
Several comments on findSpeedLimit() since Github won't let me write in there...
On next line (287), prefer to declare one variable per line. There's no need to save space like this, and it makes the declarations harder to see.  Also, spaces after the = signs (also on line 288 & 292)!
Line 302 & 307 DEBUG statements should indicate what kind of values, and even what method is writing the message. Reading the log file isn't going to be very meaningful with these vague phrases.
Readability - it's good to have a blank line prior to introducing a new logic block, esp when it begins with a comment (as most logic blocks should), like at line 309.
If block at line 310 testing for 0 speeds - since either of these conditions is possible, it is also possible that both will happen simultaneously, yet we don't handle that condition.  I'd like to see an embedded if test here that says if they're both 0 then write a warning message.
Space after comma on line 314.